tests/service_controller_test.go

```diff
@@ -38,6 +39,21 @@ var _ = Describe("Service Controller", func() {
 		Expect(serviceErr).ToNot(HaveOccurred(), "Failed to get ssp-operator-metrics service")
 	})
 
+	It("[test_id: TODO] Service ssp-operator-metrics should contain required labels", func() {
+		service, serviceErr := getSspMetricsService()
+		Expect(serviceErr).ToNot(HaveOccurred(), "Failed to get ssp-operator-metrics service")
+
+		managedByValue := service.GetLabels()[common.AppKubernetesManagedByLabel]
+		versionLabelValue := service.GetLabels()[common.AppKubernetesVersionLabel]
+		componentLabelValue := service.GetLabels()[common.AppKubernetesComponentLabel]
+		partOfValue := service.GetLabels()[common.AppKubernetesPartOfLabel]
+
+		Expect(managedByValue).To(Equal(common.AppKubernetesManagedByValue))
+		Expect(versionLabelValue).To(Equal(common.GetOperatorVersion()))
+		Expect(componentLabelValue).To(Equal(controllers.ServiceControllerName))
+		Expect(partOfValue).To(BeEmpty())
```

Comment:
I mean I'd expect it to by set when HCO will set SSP's deployment object, so then:
```
func (r *serviceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
	r.log.Info("Starting service reconciliation...", "request", req.String())
	appKubernetesPartOfValue := r.deployment.GetLabels()[common.AppKubernetesPartOfLabel]
	service := ServiceObject(req.Namespace, appKubernetesPartOfValue)
...
```
`appKubernetesPartOfValue` will be not empty.

Reply:
> One where the label is set to the value you pass to SSP

Isn't something that HCO should test? because HCO will set the part of label, and not SSP.

Reply:
No, this is a feature of SSP so SSP needs to test it.

Reply:
Sorry, I don't understand then.

1. HCO will deploy SSP [1] with the `part-of` label. This is something we can't test here.
2. If we want to set `part-of` label to SSP deployment, then we should add it to [2] (?) and then functional test can test it. This is what you mean?

[1] https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/deploy/operator.yaml#L483
[2] https://github.com/kubevirt/ssp-operator/blob/main/config/manager/manager.template.yaml#L18-L21

Reply:
To test this, you would need to add the label to the SSP deployment, and then restart the operator, by scaling it down to 0 and back up. 
It seems like too complex test just for label propagation.

Reply:
Yes, that's why I don't think it should be tested here, but maybe in HCO (if they have tests for labels at all).

Reply:
IMO, functional test should test the existing behavior of the SSP deployment, and not try to modify it and then test it, especially if that is only label - it doesn't have some critical priority in SSP deployment operation. SSP won't crash if that label will be missed.

Reply:
Agreed, the test is expensive. But this is literally a feature of SSP, so why would somebody else test this for us? A feature can be non-critical and yet it is a feature somebody could rely on, it doesn't matter if we crash or not if the feature is not working. So in order to avoid regressions we should test it.

Reply:
> I missed that the metrics service is also reconciled. It is enough to delete the service, and test that it was recreated with changed labels.

But who will add the part-of label to the SSP deployment? the functional test itself?